src/subgraph.js

```diff
@@ -39,9 +39,21 @@ const buildCombinedWarning = (filename, warnings) =>
 
 module.exports = class Subgraph {
   static async validate(data, protocol, { resolveFile }) {
+    if (protocol.name == null) {
```

Comment:
I'm trying to recall why I didn't add this error/check in the `Protocol` class constructor, I'm almost sure there's a scenario where we can't validate the `protocol` or we don't have the name 🤔 

Perhaps I'm daydreaming, restructuring all of this code and adding the whole concept of `protocol` was an adventure in itself 😅 (very fun though)

I'll try to remember it, but I think there **might** be a case where this validation shouldn't happen.

Reply:
I'll wait for your memory to wake up :) I started and put in `Protocol` first, but that was actually not given a good user experience since the validation looked more like an error than a validation.

So I moved the validation to its current location, which gives a better developer experience if protocol cannot be inferred.

This was required first because of a unit test that was failing as well as it make sense, otherwise there will be no schema to load.

Reply:
I think that was it, I'm happy that a test caught it!

I can run this branch once you fix the issues too just to be sure.